fidesctl/tests/core/test_generate_dataset.py

```diff
@@ -56,51 +117,118 @@ def test_generate_dataset_collections():
-    test_tables = ["foo", "bar"]
-
-    expected_result = {
-        "test_db": {"test_db.foo": ["1", "2"], "test_db.bar": ["3", "4"]}
-    }
-    actual_result = generate_dataset.get_db_collections_and_fields(engine)
-    assert actual_result == expected_result
+@pytest.mark.unit
+def test_unsupported_dialect_error():
+    test_url = "foo+psycopg2://fidesdb:fidesdb@fidesdb:5432/fidesdb"
+    with pytest.raises(SystemExit):
+        generate_dataset.generate_dataset(test_url, "test_file.yml")
+
+
+@pytest.mark.postgres
+class TestPostgres:
+    @pytest.fixture(scope="class", autouse=True)
+    def postgres_setup(self):
+        "Set up the Postgres Database for testing."
+        engine = sqlalchemy.create_engine(POSTGRES_URL)
+        with open("tests/data/example_sql/postgres_example.sql", "r") as query_file:
+            query = sqlalchemy.sql.text(query_file.read())
+        engine.execute(query)
+        yield
+
+    @pytest.mark.integration
+    def test_get_db_tables_postgres(self):
+        engine = sqlalchemy.create_engine(POSTGRES_URL)
+        expected_result = {
+            "public": {
+                "public.visit": ["email", "last_visit"],
+                "public.login": ["id", "customer_id", "time"],
+            }
+        }
+        actual_result = generate_dataset.get_postgres_collections_and_fields(engine)
+        assert actual_result == expected_result
+
+    @pytest.mark.integration
+    def test_generate_dataset_postgres(self):
+        actual_result = generate_dataset.generate_dataset(POSTGRES_URL, "test_file.yml")
+        assert actual_result
+
+
+@pytest.mark.mysql
+class TestMySQL:
+    @pytest.fixture(scope="class", autouse=True)
+    def mysql_setup(self):
+        """
+        Set up the MySQL Database for testing.
+
+        The query file must have each query on a separate line.
+        """
+        engine = sqlalchemy.create_engine(MYSQL_URL)
+        with open("tests/data/example_sql/mysql_example.sql", "r") as query_file:
+            queries = [query for query in query_file.read().splitlines() if query != ""]
+        print(queries)
+        for query in queries:
+            engine.execute(sqlalchemy.sql.text(query))
+        yield
+
+    @pytest.mark.integration
+    def test_get_db_tables_mysql(self):
+        engine = sqlalchemy.create_engine(MYSQL_URL)
+        expected_result = {
+            "mysql_example": {
+                "visit": ["email", "last_visit"],
+                "login": ["id", "customer_id", "time"],
+            }
+        }
+        actual_result = generate_dataset.get_mysql_collections_and_fields(engine)
+        assert actual_result == expected_result
+
+    @pytest.mark.integration
+    def test_generate_dataset_mysql(self):
+        actual_result = generate_dataset.generate_dataset(MYSQL_URL, "test_file.yml")
+        assert actual_result
+
+
+@pytest.mark.mssql
+class TestSQLServer:
+    @pytest.fixture(scope="class", autouse=True)
+    def mssql_setup(self):
```

Comment:
It also feels like this function could be part of some sort of parameterization. Maybe not a test parameter but  at least some of the code could be shared

```
        engine = sqlalchemy.create_engine(<DATABASE_URL>)
        with open("tests/data/example_sql/<DATABASE_FILE>", "r") as query_file:
            queries = [query for query in query_file.read().splitlines() if query != ""]
        print(queries)
        for query in queries:
            engine.execute(sqlalchemy.sql.text(query))
        yield
```